sdk/Sdk.Generators/FunctionMethodVisibility.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Generators
+{
+    /// <summary>
+    /// Represents the visibility of an "azure function" method and its parent classes.
+    /// </summary>
+    internal enum FunctionMethodVisibility
+    {
+        /// <summary>
+        /// The method and it's parent classes are public.
+        /// </summary>
+        PublicAndVisible,
+
+        /// <summary>
+        /// The method is public, but one or more of its parent classes are not public.
+        /// </summary>
+        PublicButContainingTypeNotVisible,
```

Comment:
nit:  `Public`, `NotPublic` and `PublicWithNonVisibleTypes` 

Or does it make more sense to have a more common enum i,.e.

`Public`, `Private`, `Internal` etc.

Reply:
Changed PublicAndVisible to Public.  I still feel "PublicButContainingTypeNotVisible" is more readable. So sticking with that.